src/rpc/server.cpp

```diff
@@ -396,9 +412,15 @@ static UniValue JSONRPCExecOne(JSONRPCRequest jreq, const UniValue& req)
 
         UniValue result = tableRPC.execute(jreq);
         rpc_result = JSONRPCReplyObj(result, NullUniValue, jreq.id);
-    }
-    catch (const UniValue& objError)
-    {
+    } catch (const int x) {
+        if (x != -1) {
+            throw;
```

Comment:
I commented in the [univalue PR](https://github.com/bitcoin-core/univalue/pull/12) that this catch should probably be for a derived class of `std::exception`, not `int`. If this is the case, there will be no need to re-throw, as the std:exception handler will catch all else below.

Reply:
Yes, throwing an int is really weird.